lib/yao/resources/role.rb

```diff
@@ -13,6 +13,33 @@ def get_by_name(name)
         self.list.find {|role| role.name == name }
       end
       alias find_by_name get_by_name
+
+      def grant(role_name, to:, on:)
+        role   = Yao::Role.get_by_name(role_name)
+        user   = Yao::User.get_by_name(to)
+        tenant = Yao::Tenant.get_by_name(on)
+
+        PUT path_for_grant_revoke(tenant, user, role)
+      end
+
+      def revoke(role_name, from:, on:)
+        role   = Yao::Role.get_by_name(role_name)
+        user   = Yao::User.get_by_name(from)
+        tenant = Yao::Tenant.get_by_name(on)
+
+        begin
+          DELETE path_for_grant_revoke(tenant, user, role)
+        rescue JSON::ParserError
+          # ParserError raises on debug mode because this OpenStack API returns
+          # empty string as response body. But this is expected behavior.
+        end
```

Comment:
mmm... How about checking `Yao.config.debug` and passing through error if not debug?